text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+For an understanding of terms utilized in this section, please see the [TUF reference](https://theupdateframework.github.io/specification/latest/).
+
+#### What is TUF?
+
+[The Update Framework](https://theupdateframework.io/) (TUF) is a software framework designed to protect software update repositories that automatically identify and download updates to software. TUF uses a series of roles and keys to provide a means to retain security, even when some keys or servers are compromised. It does this with a stated goal of requiring minimal changes and effort from repository administrators, software developers, and end users. ([Wikipedia](https://en.wikipedia.org/wiki/The_Update_Framework))
+
+TUF provides a methodology and framework for managing and verifying a modern chain of trust between what it terms "roles", stemming from a root and allowing for a further delegation of roles for different functions and paths within the repository.
+
+At the simplest level, TUF is a collection of signing key quorums which verify its own integrity, and establishes trust between the quorums and files within a repository.
+
+#### Terminology
+
+- `tuf`: The Update Framework and its specification
+- `targets`: The actual content and files distributed and to be signed
+- `repository`: One of the two TUF repositories which is where metadata for artifacts exist. This metadata is the signatures of files, snapshots, and TUF keying information.
+- `root`: The top-level set of signers/keys which are used to validation the repository
+- `delegate`: Another set of signers/keys which are used for performing specified authority within the repository
+
+## Summary & Motivations
+
+We propose the creation of two distinct TUF repositories for signing of Rust Project content and crates, respectively. Two main motivations exist for separating these concerns: The cadence of content published within each, and the trust of each. Rustup and Rust releases (both nightly and stable) are conducted under a controlled and predictable manner which is managed by the Project. However, crates are published by the community, and as such we see a larger and much more varied volume of content which may exist within this repository. We have additionally modeled signing the root of one repository by the other - this implicitly grants us a chain of trust from the "Project" (tuf-root) to the separate crates.io repository (tuf-crates). The sections below go into more detail on each repository and its configuration.
```

Comment:
I'm curious why you're modeling this as two repositories with two separate roots rather than two delegated targets which effectively namespace resources and are managed under a single TUF repo.

The idea of a TUF root is it's self-signing, so if the keys listed in that file aren't intended to be able to sign future updates to the root, perhaps it isn't a root you want.

It sounds like the idea is you want to use `tuf-root` to delegate authority to `tuf-crates`, which sounds like a delegated target to me, e.g. you could have separate targets for `crates` versus `releases` or what have you, each with authority delegated to their own independent set of keys, but with a common root able to update the keys used to manage either.